todo-app/src/ListItem.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import './App.css';
+
+ export default class ListItem extends React.Component {
+   constructor(props) {
+    super(props);
+    this.state = {
+      showEdit: true,
+      editToDo: {
+        description: props.description,
+        done: false
+      }
+    }
+
+  }
+
+  Edit(showEdit) {
+    if (!showEdit) {
+      return (
+        <div className='item-name'>
+        <form id="addItem"  onSubmit={this.handelsave.bind(this)}>
+          <input type="text" value={this.state.editToDo.description} onChange={this.handleTextChange.bind(this)}/>
+          <input type="submit" value="Save" />
+          <button type="button" onClick={this.handelTextClick.bind(this)} > Cancel </button>
+        </form>
+      </div>
+      );
+    } else {
+        return (
+          <div className='item-name'>
+          <input type="checkbox"
+            name="listItem"
+            checked={this.props.done}
+            onChange={()=>{this.props.handleCheckBoxChange(this.props.index)}}
+            />
+          <span onClick={this.handelTextClick.bind(this)}>{this.props.description} ... </span>
+          <button type="button" onClick={()=>{this.props.onDelete(this.props.index)}} > Delete </button>
+          </div>
+        )
+      }
+  }
+
+handleTextChange(event){
+  this.setState({
+    editToDo: {
+      description: event.target.value,
+      done: this.props.done
+    }
+  })
+}
+
+handelsave(event){
+  event.preventDefault()
+  this.props.save(this.state.editToDo, this.props.index)
+  this.handelTextClick()
+}
+
+handelTextClick(){
+  this.setState(prevState => ({
+      showEdit: !prevState.showEdit
```

Comment:
Nice 👍

But this can lead to confusion. Whenever we want to change the edit state of the `ListItem` component we call `handleTextClick` and we assume that we get the correct result. Would it make it more clear to rename the function according to its scope, pass the value you want directly to the function

```javascript
setShowEdit (showEdit) {
  this.setState({ showEdit }); // note here we are using the object property shorthand
});
```

and use it like `this.setShowEdit(true)` or `this.setShowEdit(false)`? What do you think?